packages/devtools_app/lib/src/shared/http/http_request_data.dart

```diff
@@ -46,6 +48,99 @@ class DartIOHttpRequestData extends NetworkRequest {
     }
   }
 
+  factory DartIOHttpRequestData.fromJson(Map<String, dynamic> requestData) {
```

Comment:
instead of putting all this HAR specific code on the `DartIOHttpRequestData` class, can you  move it to the HarNetworkData class? My concern is that we are blurring the lines of where the Har format begins and ends within the DevTools codebase, and it would be preferable to keep all this Har-specific code with the Har-specific data classes.

It could be reasonable to create another class `HarDataEntry` in `har_network_data.dart` where you have methods `.fromDartIOHttpRequest` and `.toDartIOHttpRequest`. This will keep the Har data encoding separate from the DevTools network data structures.

Reply:
I have created HarDataEntry class and added implementation for from json,
for tojson
I have added implementation in a separate pr for demonstration -> https://github.com/hrajwade96/devtools/pull/2
Please check and let me know if the structure looks ok

Reply:
@kenzieschmoll pls clarify if above change is required or it is good as it is

Reply:
I left a comment [here](https://github.com/flutter/devtools/pull/7970#discussion_r1684547365) regarding the toJson implementation. If you can, it would be best to add all your changes to this PR so we can keep the review in a single place.

Reply:
done

Reply:
Sure, will add those changes here itself